hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -1868,12 +1868,12 @@ public boolean isDefinedOnSubclass(int i) {
 	}
 
 	@Override
-	public int[] resolveAttributeIndexes(Set<String> properties) {
-		Iterator<String> iter = properties.iterator();
-		int[] fields = new int[properties.size()];
+	public int[] resolveAttributeIndexes(String[] attributeNames) {
+		int[] fields = new int[attributeNames.length];
```

Comment:
And yes... I realize our specific code does make sure the array entries are unique.  But this is part of a public SPI